### ISSUE TITLE
messages_11.validate_xml() incorrectly fails for certain inputs

### ISSUE THREAD

**Issue by MarkDavidson** [TAXIIProject/libtaxii#100]
This is a continuation of: https://github.com/crits/crits_services/issues/42

Currently, calling `tm11.validate_xml()` on this TAXII Poll Response incorrectly results in an error:

``` xml
<taxii_11:Poll_Response xmlns:taxii_11="http://taxii.mitre.org/messages/taxii_xml_binding-1.1" message_id="99413" in_response_to="45010" collection_name="default" more="false">
  <taxii_11:Inclusive_End_Timestamp>2014-08-14T17:54:00.188214+00:00</taxii_11:Inclusive_End_Timestamp>
  <taxii_11:Record_Count partial_count="false">2</taxii_11:Record_Count>
  <taxii_11:Content_Block>
    <taxii_11:Content_Binding binding_id="urn:stix.mitre.org:xml:1.1.1"/>
    <taxii_11:Content>
      <stix:STIX_Package xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:stix="http://stix.mitre.org/stix-1" xmlns:indicator="http://stix.mitre.org/Indicator-2" xmlns:cybox="http://cybox.mitre.org/cybox-2" xmlns:DomainNameObj="http://cybox.mitre.org/objects#DomainNameObject-1" xmlns:cyboxVocabs="http://cybox.mitre.org/default_vocabularies-2" xmlns:stixVocabs="http://stix.mitre.org/default_vocabularies-1" xmlns:example="http://example.com/" xsi:schemaLocation="http://stix.mitre.org/stix-1 ../stix_core.xsd     http://stix.mitre.org/Indicator-2 ../indicator.xsd     http://cybox.mitre.org/default_vocabularies-2 ../cybox/cybox_default_vocabularies.xsd     http://stix.mitre.org/default_vocabularies-1 ../stix_default_vocabularies.xsd     http://cybox.mitre.org/objects#DomainNameObject-1 ../cybox/objects/Domain_Name_Object.xsd" id="example:STIXPackage-f61cd874-494d-4194-a3e6-6b487dbb6d6e" timestamp="2014-05-08T09:00:00.000000Z" version="1.1.1">
    <stix:STIX_Header>
        <stix:Title>Example watchlist that contains domain information.</stix:Title>
        <stix:Package_Intent xsi:type="stixVocabs:PackageIntentVocab-1.0">Indicators - Watchlist</stix:Package_Intent>
    </stix:STIX_Header>
    <stix:Indicators>
        <stix:Indicator xsi:type="indicator:IndicatorType" id="example:Indicator-2e20c5b2-56fa-46cd-9662-8f199c69d2c9" timestamp="2014-05-08T09:00:00.000000Z">
            <indicator:Type xsi:type="stixVocabs:IndicatorTypeVocab-1.1">Domain Watchlist</indicator:Type>
            <indicator:Description>Sample domain Indicator for this watchlist</indicator:Description>
            <indicator:Observable id="example:Observable-87c9a5bb-d005-4b3e-8081-99f720fad62b">
                <cybox:Object id="example:Object-12c760ba-cd2c-4f5d-a37d-18212eac7928">
                    <cybox:Properties xsi:type="DomainNameObj:DomainNameObjectType" type="FQDN">
                        <DomainNameObj:Value condition="Equals" apply_condition="ANY">malicious1.example.com##comma##malicious2.example.com##comma##malicious3.example.com</DomainNameObj:Value>
                    </cybox:Properties>
                </cybox:Object>
            </indicator:Observable>
        </stix:Indicator>
    </stix:Indicators>
</stix:STIX_Package>
    </taxii_11:Content>
    <taxii_11:Timestamp_Label>2014-03-10T13:27:04.455445+00:00</taxii_11:Timestamp_Label>
  </taxii_11:Content_Block>
</taxii_11:Poll_Response>
```

Error message:
`<string>:7:0:ERROR:SCHEMASV:SCHEMAV_CVC_ELT_4_2: Element '{http://stix.mitre.org/stix-1}Indicator', attribute '{http://www.w3.org/2001/XMLSchema-instance}type': The QName value '{http://stix.mitre.org/Indicator-2}IndicatorType' of the xsi:type attribute does not resolve to a type definition.`  

There's a few considerations here:
- The `Content` element in `Content_Block` allows `anyMixedContent` with `processContents="lax"`  
  - I think this means that lxml and/or libxml2 might not be doing the right thing. I don't know why a failure to resolve a schema element is resulting in a validation failure when lax means "if the schema cannot be obtained, no errors will occur." (http://msdn.microsoft.com/en-us/library/ms256043(v=vs.110).aspx)
- Therefore, I believe a tactical solution is in order. Either import the STIX schemas or change processContents to `processContents="skip"`.
  - Importing the STIX schemas would force everyone to pay the overhead of loading the STIX schemas for every validation, regardless of whether or not they are using STIX (a negative, IMO)
  - I can only import one version of the STIX schemas (e.g., STIX 1.1.1, 1.1, 1.0.1, or 1.0) and that would leave people using other STIX versions with XML they can't validate (a negative, IMO)
  - In changing `processContents="skip"`, no Content fields will be validated (by schema), but other mechanisms could be used in software (e.g., pull out the STIX portion of the XML and perform validation outside of libtaxii). (a negative, IMO)
- Side note:I briefly explored XML Catalogs in libxml2 (documentation here: http://xmlsoft.org/catalog.html), but I couldn't get them to work (I couldn't tell if my catalog was loading or not, but nothing I did seemed to have any effect)
- I think changing `processContents="skip"` is the least bad (though in reality I'd like to find out whether this is an error in the schema we've created for TAXII or a processing error in lxml/libxml2).

**TL:DR;** I'm probably going to change the value of `processContents` to `skip` instead of `lax` in the TAXII Schema that comes with libtaxii to circumvent this issue altogether, though this is probably worth revisiting in the future to see if it's the correct solution or not.
